packages/router/utils.ts

```diff
@@ -702,6 +708,21 @@ function compilePath(
   return [matcher, paramNames];
 }
 
+function safelyDecodeURI(value: string) {
+  try {
+    return decodeURI(value);
```

Comment:
How sure are you that `decodeURI` is what we want here? It's been a while since I dug into the URL encoding stuff, but last time I did I remember reading something about "you almost always want decodeURIComponent, not decodeURI". Does that sound familiar?

Reply:
I'll do some digging but I think that would work.  The choice of `decodeURI` instead of `decodeURIComponent` was to align with when I was initially encoding via `encodeURI` instead of `encodeURIComponent` (since `encodeURIComponent` is a bit too aggressive).  

But now that encoding is done via `new URL` it may be fine to switch to `decodeURIComponent`

<img width="374" alt="Screen Shot 2022-10-20 at 3 37 33 PM" src="https://user-images.githubusercontent.com/1609022/197042050-0a0622b5-dd72-4102-a3fb-a2b3ba2b2266.png">

Reply:
I think want to keep `decodeURI` here - since `decodeURIComponent` is also a bit too aggressive and _also_ decodes properly-encoded URL chars such as `/` and `?` whereas decodeURI will keep them encoded.  See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#description for an explanation of the treatment of special meaning URL characters.

We actually have a test around this too in `params-decode-test` for when a `/` is used in a URL like `/content/%2F`.  Here's a quick example of how `decodeURIComponent` goes too far in this case and creates a path with unintended slash separators and query strings:

<img width="567" alt="Screen Shot 2022-10-20 at 4 10 30 PM" src="https://user-images.githubusercontent.com/1609022/197048243-dab34c48-3e1b-4c5e-bd2b-df50356f14bb.png">

Reply:
Sounds good. Thanks for digging 👍